src/lib/header-bidding/a9/a9.ts

```diff
@@ -70,28 +98,68 @@ const requestBids = async (
 		return requestQueue;
 	}
 
-	requestQueue = requestQueue
-		.then(
-			() =>
-				new Promise<void>((resolve) => {
-					window.apstag?.fetchBids({ slots: adUnits }, () => {
-						window.googletag.cmd.push(() => {
-							window.apstag?.setDisplayBids();
-							resolve();
-						});
-					});
-				}),
-		)
-		.catch(() => {
-			reportError(new Error('a9 header bidding error'), 'commercial');
-		});
+	const updatedAdUnits = adUnits.map((adUnit) => {
+		return {
+			...adUnit,
+			blockedBidders: getBlockBidders(adUnit),
+		};
+	});
+
+	const { blockedBidderAdUnits, nonBlockedBidderAdUnits } =
+		updatedAdUnits.reduce<{
+			blockedBidderAdUnits: A9AdUnitExtended[];
+			nonBlockedBidderAdUnits: A9AdUnitExtended[];
+		}>(
+			(result, adUnit) => {
+				const hasBlockedBidders = adUnit.blockedBidders.length > 0;
+
+				if (hasBlockedBidders) {
+					result.blockedBidderAdUnits.push(adUnit);
+				} else {
+					result.nonBlockedBidderAdUnits.push(adUnit);
+				}
+
+				return result;
+			},
+			{ blockedBidderAdUnits: [], nonBlockedBidderAdUnits: [] },
+		);
+
+	const groupedAdUnits = [
+		{ adUnits: blockedBidderAdUnits, blockedBidders: ['1lsxjb4'] },
+		{ adUnits: nonBlockedBidderAdUnits, blockedBidders: [] },
+	];
```

Comment:
I think this could be done with a bit less code

```suggestion
	const blockGumGumAdUnits = adunits.filter((adUnit) => getBlockedBidders(adUnit));
	const allowGumGumAdUnits = adunits.filter((adUnit) => !getBlockedBidders(adUnit));

	
	const groupedAdUnits = [
		{ adUnits: blockGumGumAdUnits, blockedBidders: ['1lsxjb4'] },
		{ adUnits: allowGumGumAdUnits, blockedBidders: [] },
	];
```
`getBlockedBidders` could be `shouldBlockGumGum` or something instead also.

Reply:
@Jakeii @cemms1 I also refactored to `shouldBlockGumGum` to return a boolean since the addition of the appropriate array is added as part of the grouped objects in `groupedAdUnits`